docs/docsite/rst/conf.py

```diff
@@ -386,3 +392,22 @@
 ]
 linkcheck_workers = 25
 # linkcheck_anchors = False
+
+# Generate redirects for pages when building on Read The Docs
+def setup(app: Sphinx) -> dict[str, str]:
+
+    if tags.has('redirects'):
+
+        redirects = {}
+        redirects_config_path = DOCS_ROOT_DIR.parent / "ansible_redirects.toml"
+        with open(redirects_config_path, "rb") as f:
+            redirects = tomllib.load(f)
+
+        app.setup_extension('sphinx_reredirects') # redirect pages that have been restructured or removed
+        app.config.redirects = redirects
```

Comment:
@webknjaz You mentioned in another comment that modifying the configuration after loading the extension seems problematic. Can you take another look at this please? I'm not sure I see another way to do that or fully understand how this could pose an issue.

Reply:
@oraNod when you call `app.setup_extension()`, I think it may invoke some hooks from that extension at that point. And so those hooks won't get this configuration value, accessing the config object before it's assigned. I don't know if it will have any effect in our case. If it works, then fine. Leave it be.